src/hash.ts

```diff
@@ -148,3 +115,39 @@ export async function initHasher(): Promise<unknown> {
   }
   return hasherPromise;
 }
+
+let tempHashCounter = 0;
+
+const tempPrefix = '/t/';
+
+export function newTempHash(): Hash {
+  // Must not overlap with Hash.prototype.toString results
+  return (tempPrefix +
+    (tempHashCounter++)
+      .toString()
+      .padStart(HASH_LENGTH - tempPrefix.length, '0')) as unknown as Hash;
+}
+
+export function isHash(v: unknown): v is Hash {
```

Comment:
newTempHash returns something of type Hash,

but 

```
isHash(newTempHash()) === false
```

which seems wrong.

Reply:
Agreed. Let me think a bit

Reply:
I made isHash include isTempHash

I also added assertNotTempHash before writing hashes to the underlying KV store.